glide-core/src/client/value_conversion.rs

```diff
@@ -287,6 +288,27 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         },
+        ExpectedReturnType::KeyWithMemberAndScore => match value {
+            Value::Nil => Ok(value),
+            Value::Array(ref array) if array.len() == 3 && matches!(array[2], Value::Double(_)) => {
+                Ok(value)
+            }
+            Value::Array(mut array)
+                if array.len() == 3
+                    && matches!(array[2], Value::BulkString(_) | Value::SimpleString(_)) =>
+            {
+                array[2] =
+                    convert_to_expected_type(array[2].clone(), Some(ExpectedReturnType::Double))
+                        .unwrap();
```

Comment:
Why unwrap here? This looks like a recursive call, so you could just `?` the error up to the caller.

Reply:
I'm not sure I understand, maybe because I'm still a little hazy with Rust code. I tried removing `.unwrap()` but I get a compilation error: 
```
convert_to_expected_type(array[2].clone(), Some(ExpectedReturnType::Double));
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Value`, found `Result<Value, RedisError>`
```

Reply:
`unwrap` panics, which crashes the application. This is good for cases where there is a bug in the code due to some invariant being broken where you don't want to continue running the code in a bad state. You don't want to use it for errors you can recover from. In this case, this function already returns a `Result` which is used for recoverable errors. So you can just use this `?` to return the error:
```suggestion
                    convert_to_expected_type(array[2].clone(), Some(ExpectedReturnType::Double))
                        .unwrap()?;
```

Reply:
Got it, thanks, updated now